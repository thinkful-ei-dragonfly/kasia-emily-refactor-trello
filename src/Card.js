import React from 'react';



export default function(props) {
  return (
    <div className="Card">
      <button type="button" onClick={() => props.handleClick(props.listId, props.id)} >delete</button> 
      <h3>{props.title}</h3>
      <p>{props.content}</p> 
    </div>
  );
}