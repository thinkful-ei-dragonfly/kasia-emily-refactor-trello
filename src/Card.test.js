import React from 'react';
import ReactDOM from 'react-dom';
//reactDOM = tool used to render component
import renderer from 'react-test-renderer';
//^turns html into json
import Card from './Card';



describe('Card component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    //div is standard placeholder, can be anything
    ReactDOM.render(<Card key={4} title={'hello'} content={'hello'}/>,div);
    //^this is the important line
    ReactDOM.unmountComponentAtNode(div);
    //stop component from being rendered in the div 
  });

  it('renders the UI as expexted', () => {
    const tree = renderer
      .create(<Card key={2} title={'hello'} content={'hello'}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});





//smoke test
  //