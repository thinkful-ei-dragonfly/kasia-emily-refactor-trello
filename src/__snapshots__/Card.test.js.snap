// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expexted 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    hello
  </h3>
  <p>
    hello
  </p>
</div>
`;
